client/state/selectors/is-eligible-for-checkout-to-checklist.js

```diff
@@ -10,26 +10,31 @@ import {
 	hasTransferProduct,
 	hasPlan,
 	hasConciergeSession,
-	hasEcommercePlan,
 } from 'lib/cart-values/cart-items';
 import isEligibleForDotcomChecklist from './is-eligible-for-dotcom-checklist';
+import { retrieveSignupDestination } from 'signup/utils';
 
 /**
  * @param {Object} state Global state tree
  * @param {Number} siteId Site ID
  * @param {Object} cart object
+ * @param {String} destination The final destination URL after signup completes
  * @return {Boolean} True if current user is able to see the checklist after checkout
  */
-export default function isEligibleForCheckoutToChecklist( state, siteId, cart ) {
+export default function isEligibleForSignupDestination( state, siteId, cart ) {
```

Comment:
The destination is no longer just Checklist, so it makes sense to rename the function to avoid confusion,